lnd_test.go

```diff
@@ -2000,34 +2054,67 @@ func testChannelForceClosure(net *lntest.NetworkHarness, t *harnessTest) {
 	if err != nil {
 		t.Fatalf("unable to generate block: %v", err)
 	}
-	time.Sleep(duration)
+
+	// Due to mining a large amount of blocks, we'll sleep to allow the UTXO
+	// nursery to catch up before restart.
+	time.Sleep(3 * time.Second)
 
 	// We now restart Alice, to ensure that she will broadcast the presigned
 	// htlc timeout txns after the delay expires after experiencing a while
 	// waiting for the htlc outputs to incubate.
 	if err := net.RestartNode(net.Alice, nil); err != nil {
 		t.Fatalf("Node restart failed: %v", err)
 	}
-	time.Sleep(duration)
 
-	pendingChanResp, err = net.Alice.PendingChannels(ctxb, pendingChansRequest)
-	if err != nil {
-		t.Fatalf("unable to query for pending channels: %v", err)
-	}
-	assertNumForceClosedChannels(t, pendingChanResp, 1)
+	err = lntest.WaitPredicate(func() bool {
```

Comment:
These wait predicates are quite big. I think it would make sense to add a comment above each of them, summarizing what they wait for.

Reply:
Fixed.